src/Module/Admin/Federation.php

```diff
@@ -0,0 +1,200 @@
+		// social and statusnet nodes this node is knowing
+		//
+		// We are looking for the following platforms in the DB, "Red" should find
+		// all variants of that platform ID string as the q() function is stripping
+		// off one % two of them are needed in the query
+		// Add more platforms if you like, when one returns 0 known nodes it is not
+		// displayed on the stats page.
+		$platforms = ['Friendi%%a', 'Diaspora', '%%red%%', 'Hubzilla', 'BlaBlaNet', 'GNU Social', 'StatusNet', 'Mastodon', 'Pleroma', 'socialhome', 'ganggo'];
+		$colors = [
+			'Friendi%%a' => '#ffc018', // orange from the logo
+			'Diaspora'   => '#a1a1a1', // logo is black and white, makes a gray
+			'%%red%%'    => '#c50001', // fire red from the logo
+			'Hubzilla'   => '#43488a', // blue from the logo
+			'BlaBlaNet'  => '#3B5998', // blue from the navbar at blablanet-dot-com
+			'GNU Social' => '#a22430', // dark red from the logo
+			'StatusNet'  => '#789240', // the green from the logo (red and blue have already others
+			'Mastodon'   => '#1a9df9', // blue from the Mastodon logo
+			'Pleroma'    => '#E46F0F', // Orange from the text that is used on Pleroma instances
+			'socialhome' => '#52056b', // lilac from the Django Image used at the Socialhome homepage
+			'ganggo'     => '#69d7e2', // from the favicon
+		];
+		$counts = [];
+		$total = 0;
+		$users = 0;
+
+		foreach ($platforms as $platform) {
+			// get a total count for the platform, the name and version of the
+			// highest version and the protocol tpe
+			$platformCountStmt = DBA::p('SELECT
+       			COUNT(*) AS `total`,
+       			SUM(`registered-users`) AS `users`,
+       			ANY_VALUE(`platform`) AS `platform`,
+				ANY_VALUE(`network`) AS `network`,
+       			MAX(`version`) AS `version` FROM `gserver`
+				WHERE `platform` LIKE ?
+			  	AND `last_contact` >= `last_failure`
+				ORDER BY `version` ASC', $platform);
+			$platformCount = DBA::fetch($platformCountStmt);
+			$total += $platformCount['total'];
+			$users += $platformCount['users'];
+
+			DBA::close($platformCountStmt);
+
+			// what versions for that platform do we know at all?
+			// again only the active nodes
+			$versionCountsStmt = DBA::p('SELECT
+       			COUNT(*) AS `total`,
+       			`version` FROM `gserver`
+				WHERE `last_contact` >= `last_failure`
+				AND `platform` LIKE ?
+				GROUP BY `version`
+				ORDER BY `version`;', $platform);
+			$versionCounts = DBA::toArray($versionCountsStmt);
+
+			//
+			// clean up version numbers
+			//
+			// some platforms do not provide version information, add a unkown there
+			// to the version string for the displayed list.
+			foreach ($versionCounts as $key => $value) {
+				if ($versionCounts[$key]['version'] == '') {
+					$versionCounts[$key] = ['total' => $versionCounts[$key]['total'], 'version' => L10n::t('unknown')];
+				}
+			}
+
+			// Reformat and compact version numbers
+			if ($platform == 'Pleroma') {

```

Comment:
The Pleroma version number handling contains a regression towards current behaviour in the development branch. as all the `-` version numbers are listed again, instead of the `x.x.x` version.

Reply:
Good catch!